spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/profile/AbstractKubernetesProfileEnvironmentPostProcessor.java

```diff
@@ -60,72 +58,64 @@
 	@Override
 	public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
 
-		final boolean kubernetesEnabled = environment.getProperty("spring.cloud.kubernetes.enabled", Boolean.class,
-				true);
+		application.addInitializers(ctx -> LOG.replayTo(AbstractKubernetesProfileEnvironmentPostProcessor.class));
+
+		boolean kubernetesEnabled = environment.getProperty("spring.cloud.kubernetes.enabled", Boolean.class, true);
 		if (!kubernetesEnabled) {
 			return;
 		}
 		addNamespaceFromServiceAccountFile(environment);
-		if (isInsideKubernetes(environment)) {
-			if (hasKubernetesProfile(environment)) {
-				if (LOG.isDebugEnabled()) {
-					LOG.debug("'kubernetes' already in list of active profiles");
-				}
-			}
-			else {
-				if (LOG.isDebugEnabled()) {
-					LOG.debug("Adding 'kubernetes' to list of active profiles");
-				}
-				environment.addActiveProfile(KUBERNETES_PROFILE);
-			}
-		}
-		else {
-			if (LOG.isDebugEnabled()) {
-				LOG.warn("Not running inside kubernetes. Skipping 'kubernetes' profile activation.");
-			}
-		}
+		addKubernetesProfileIfMissing(environment);
 	}
 
 	protected abstract boolean isInsideKubernetes(Environment environment);
 
 	private boolean hasKubernetesProfile(Environment environment) {
-		for (String activeProfile : environment.getActiveProfiles()) {
-			if (KUBERNETES_PROFILE.equalsIgnoreCase(activeProfile)) {
-				return true;
-			}
-		}
-		return false;
+		return Arrays.stream(environment.getActiveProfiles()).anyMatch(KUBERNETES_PROFILE::equalsIgnoreCase);
 	}
 
 	@Override
 	public int getOrder() {
 		return ORDER;
 	}
 
-	private void addNamespaceFromServiceAccountFile(ConfigurableEnvironment environment) {
-		Map<String, Object> properties = new HashMap<>();
+	private void addKubernetesProfileIfMissing(ConfigurableEnvironment environment) {
+		if (isInsideKubernetes(environment)) {
+			if (hasKubernetesProfile(environment)) {
+				LOG.debug("'kubernetes' already in list of active profiles");
+			}
+			else {
+				LOG.debug("Adding 'kubernetes' to list of active profiles");
+				environment.addActiveProfile(KUBERNETES_PROFILE);
+			}
+		}
+		else {
+			LOG.warn("Not running inside kubernetes. Skipping 'kubernetes' profile activation.");
+		}
+	}
 
+	private void addNamespaceFromServiceAccountFile(ConfigurableEnvironment environment) {
 		String serviceAccountNamespace = environment.getProperty(NAMESPACE_PATH_PROPERTY,
 				SERVICE_ACCOUNT_NAMESPACE_PATH);
-		LOG.debug("Looking for service account namespace at " + serviceAccountNamespace);
-		boolean serviceAccountNamespaceExists = Files.isRegularFile((new File(serviceAccountNamespace)).toPath(),
```

Comment:
- `new File(serviceAccountNamespace)).toPath()` can be simplified to `Paths.get(serviceAccountNamespace)`
- `new LinkOption[0]` is not needed to be provided explicitly, this is done implicitly already 
- added `[` and `]` in logger message, to be consistent across every other logger that includes `serviceAccountNamespace`
-